### PR TITLE
manages NodeConfig objects within the guest cluster's namespace

### PR DESCRIPTION
**Pull request by xh3b4sd**
Towards https://github.com/giantswarm/giantswarm/issues/2935.

### CODE REVIEW

File: service/awsconfig/v10/resource/lifecycle/create.go [hunk 1]
```diff
@@ -121,7 +121,7 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 				return microerror.Mask(err)
 			}
 
-			n := v1.NamespaceDefault
+			n := customObject.GetNamespace()
```
Comment (xh3b4sd) [inline service/awsconfig/v10/resource/lifecycle/create.go]: When fetching a `NodeConfig` initially we use the namespace of the `AWSConfig`. Same for creating the `NodeConfig` if it is not there yet.

File: service/awsconfig/v10/resource/lifecycle/create.go [hunk 2]
```diff
@@ -146,7 +146,7 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 	{
 		r.logger.LogCtx(ctx, "level", "debug", "message", "start inspection of node configs for the guest cluster")
 
-		n := v1.NamespaceDefault
+		n := v1.NamespaceAll
```
Comment (xh3b4sd) [inline service/awsconfig/v10/resource/lifecycle/create.go]: When listing we go with all namespaces to get the full list for inspection below.

File: service/awsconfig/v10/resource/lifecycle/create.go [hunk 3]
```diff
@@ -258,11 +258,11 @@ func (r *Resource) createNodeConfig(ctx context.Context, customObject providerv1
...
 }
 
-func (r *Resource) deleteNodeConfig(ctx context.Context, privateDNS string) error {
+func (r *Resource) deleteNodeConfig(ctx context.Context, nodeConfig corev1alpha1.NodeConfig) error {
 	r.logger.LogCtx(ctx, "level", "debug", "message", "deleting node config for guest cluster node")
 
-	n := v1.NamespaceDefault
-	i := privateDNS
+	n := nodeConfig.GetNamespace()
+	i := nodeConfig.GetName()
```
Comment (xh3b4sd) [inline service/awsconfig/v10/resource/lifecycle/create.go]: When deleting we use the namespace and name provided by the object.

File: service/awsconfig/v10/resource/lifecycle/create.go [hunk 4]
```diff
@@ -8,7 +8,7 @@ import (
 	"github.com/aws/aws-sdk-go/service/autoscaling"
 	"github.com/aws/aws-sdk-go/service/cloudformation"
 	"github.com/aws/aws-sdk-go/service/ec2"
-	"github.com/giantswarm/apiextensions/pkg/apis/core/v1alpha1"
+	corev1alpha1 "github.com/giantswarm/apiextensions/pkg/apis/core/v1alpha1"
```
Comment (xh3b4sd) [inline service/awsconfig/v10/resource/lifecycle/create.go]: Just boyscouting to align with the other apiextensions import.

File: service/awsconfig/v10/resource/lifecycle/create.go (hunk 3 above)
Comment (kopiczko) [reply service/awsconfig/v10/resource/lifecycle/create.go]: We should do it always actually.

**Pull request merged**